pkg/tcpip/transport/tcp/cubic.go

```diff
@@ -66,6 +118,56 @@ func (c *cubicState) enterCongestionAvoidance() {
 	}
 }
 
+// updateHyStart tracks packet round-trip time (rtt) to find a safe threshold
+// to exit slow start without triggering packet loss.  It updates the SSThresh
+// and sets FoundThresh when it does.
+func (c *cubicState) updateHyStart(rtt time.Duration) {
+	if rtt < 0 {
+		// negative indicates unknown
+		return
+	}
+	now := c.s.ep.stack.Clock().NowMonotonic()
+	if c.EndSeq.LessThan(c.s.SndUna) {
+		c.beginHyStartRound(now)
+	}
+	// ACK train
+	if now.Sub(c.LastAck) < ackDelta && // ensures acks are part of the same "train"
+		c.LastRTT < effectivelyInfinity {
+		c.LastAck = now
+		thresh := c.LastRTT / 2
+		if now.Sub(c.RoundStart) > thresh {
+			c.s.Ssthresh = c.s.SndCwnd
+		}
+	}
+
+	// Delay increase
+	c.CurrRTT = minDuration(c.CurrRTT, rtt)
+	c.SampleCount++
+
+	if c.SampleCount >= nRTTSample &&
+		c.LastRTT < effectivelyInfinity {
```

Comment:
Optional/style: the `if` condition can be one line if you prefer.

Reply:
I find it easier to read complex boolean equations with the clauses on different lines, but I'm happy to change if that's the preferred style for this project.